org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/MarkupContentFactory.java

```diff
@@ -57,4 +59,25 @@ public static MarkupContent createMarkupContent(String value, String preferredKi
 		}
 		return content;
 	}
+
+	/**
+	 * Create the markup content according the given markup kind and the capability
+	 * of the client.
+	 * 
+	 * @param values         the list of documentation values
+	 * @return the markup content according the given markup kind and the capability
+	 *         of the client.
+	 */
+	public static MarkupContent creatMarkupContent(List<String> values, IMarkupKindSupport request) {
+		String kind = request.canSupportMarkupKind(MarkupKind.MARKDOWN) ? MarkupKind.MARKDOWN : MarkupKind.PLAINTEXT;
+		if (values.size() ==1) {
+			return new MarkupContent(kind, values.get(0));
+		}
+		StringBuilder retValue = new StringBuilder();
+		for (String value : values) {
+			retValue.append(value);
+			retValue.append("___");

```

Comment:
What about with plaintext? I think we should generate that only for markdown and generate a System.lineSeparator() in another case.

Reply:
Sorry just saw your reply.  I will update thid PR.

Reply:
PR has been updated.